Aerial: add port

danchr opened the pull request:
#### Description

This is a new port. I'm doing this via a PR for review; I had to do some horrible trickery with relocating the build to `/tmp` due to permission errors. Is that acceptable?

See also https://trac.macports.org/ticket/57234

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Generate version information with this command in shell:
    echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)"; echo "Xcode $(xcodebuild -version | awk -v ORS=' ' '{print $NF}')"
-->
macOS 10.x
Xcode 8.x

###### Verification <!-- (delete not applicable items) -->
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

ryandesign commented on aqua/Aerial/Portfile:
```diff
@@ -0,0 +1,84 @@
+# although the binary works on 10.9 and later, it doesn't build on
+# anything earlier than 10.15
+pre-fetch {
+    if {${os.subplatform} eq "macosx" && [vercmp ${macosx_version} 10.15] < 0} {
```
Please also set `known_fail yes` when this condition is true, so that the buildbot will know to skip the build. Move the condition to the top level and move the pre-fetch block inside it.

ryandesign commented:
Looks like the build fails with Xcode 12 beta:
```
error: cannot find type 'NSTintConfiguration' in scope
```

ryandesign commented:
> I had to do some horrible trickery with relocating the build to `/tmp` due to some issues with trace mode. Is that acceptable?

I've don't recall any other port needing to do that, so it would certainly be unusual. Can you provide more detail on the problems you saw?

danchr commented:
> Looks like the build fails with Xcode 12 beta:
> 
> ```
> error: cannot find type 'NSTintConfiguration' in scope
> ```

Yeah, I incorrectly assumed that one of the patches was dependent on the Xcode version rather than the OS version; fixed!

> > I had to do some horrible trickery with relocating the build to `/tmp` due to some issues with trace mode. Is that acceptable?
> 
> I don't recall any other port needing to do that, so it would certainly be unusual. Can you provide more detail on the problems you saw?

One possible cause is that I put `/opt` on a separate APFS container on both machines where I tested this. I've added a commit that moves the build directory into `${workpath}` and triggers the error for me. I've pasted the final output below; could you perhaps try the branch and see if you get a similar error?

```
dyld: warning: could not load inserted library '/opt/local/libexec/macports/lib/darwintrace1.0/darwintrace.dylib' into hardened process because no suitable image found.  Did find:
	/opt/local/libexec/macports/lib/darwintrace1.0/darwintrace.dylib: code signature in (/opt/local/libexec/macports/lib/darwintrace1.0/darwintrace.dylib) not valid for use in process using Library Validation: mapping process is a platform binary, but mapped file is not
	/opt/local/libexec/macports/lib/darwintrace1.0/darwintrace.dylib: stat() failed with errno=1
2020-08-31 07:40:41.466 xcodebuild[54711:15936422] [MT] IDELogStore: Failed to open log store at /opt/local/var/macports/build/_Users_dan_src_opensrc_os_macports-ports_aqua_Aerial/Aerial/work/Aerial-2.0.5/build/Logs/Test
2020-08-31 07:40:41.467 xcodebuild[54711:15936422] [MT] IDELogStore: Failed to open Test log store: Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “LogStoreManifest.plist” in the folder “Test”." UserInfo={NSURL=file:///opt/local/var/macports/build/_Users_dan_src_opensrc_os_macports-ports_aqua_Aerial/Aerial/work/Aerial-2.0.5/build/Logs/Test/LogStoreManifest.plist, NSUserStringVariant=Folder, NSUnderlyingError=0x7f80727cbc50 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}. User info: {
    NSURL = "file:///opt/local/var/macports/build/_Users_dan_src_opensrc_os_macports-ports_aqua_Aerial/Aerial/work/Aerial-2.0.5/build/Logs/Test/LogStoreManifest.plist";
    NSUnderlyingError = "Error Domain=NSPOSIXErrorDomain Code=1 \"Operation not permitted\"";
    NSUserStringVariant = Folder;
}.
Build Preparation
Couldn't create workspace arena folder '/opt/local/var/macports/build/_Users_dan_src_opensrc_os_macports-ports_aqua_Aerial/Aerial/work/Aerial-2.0.5/build': Unable to write to info file '<DVTFilePath:0x7f80727cc520:'/opt/local/var/macports/build/_Users_dan_src_opensrc_os_macports-ports_aqua_Aerial/Aerial/work/Aerial-2.0.5/build/info.plist'>'.
```

Interestingly, some files _do_ get written in there, but the build still fails; creating the files and retrying doesn't appear to help.

danchr commented:
Here's a full log:

[aerial.log](https://github.com/macports/macports-ports/files/5148536/aerial-log.txt)

danchr commented:
I managed to isolate it! The root issue is that `/opt` is on a separate volume, and this leads to some Xcode tools accessing `/opt/.TemporaryItems`, which is denied by the sandbox. Hardcoding this directory to allowed in `/opt/local/libexec/macports/lib/port1.0/portsandbox.tcl` allows the port to build; all I have to figure out now is how to properly detect that this is needed from within Tcl…

mascguy commented on aqua/Aerial/Portfile:
```diff
@@ -0,0 +1,87 @@
+set custom_build_dir    ${worksrcpath}/build
+
+xcode.build.settings-append \
+                        BUILD_DIR=${custom_build_dir}
```
Dan/Anyone, we're seeing buildbot failures on 10.11 for `MenuMeters`, with the classic symptoms related to CoreSimulator, etc. However, it looks like this was fixed for base, via [PR 208 - sandbox: handle /opt/local on separate mountpoint](https://github.com/macports/macports-base/pull/208).

Would you folks expect this issue to still occur on our buildbot?

https://build.macports.org/builders/ports-10.11_x86_64-builder/builds/160675/steps/install-port/logs/stdio

mascguy replied:
And if the fix for base isn't enough, would we want to try the same fix that was tried here (relative to setting `BUILD_DIR`, etc)?

CC: @Tatsh

ryandesign replied:
> it's actually a compile error

Right:

```
MenuMetersMenuExtraBase.m:34:8: error: unexpected '@' in program
    if(@available(macOS 10.14,*)){
       ^
```
